docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,492 @@
+        * **clientKey**: defines a client private key.
+          * one of `inline`, `inlineString` or `secret`.
+* **status**: status of an object managed by Kuma control-plane
+  * **vip**: section for allocated IP
+    * **ip**: allocated IP for a provided domain with `InternalVIP` type in a match section
+    * **type**: provides information about the way IP was provided
+  * **addresses**: list of domains and addresses
+    * **address**: IP address, CIDR, or domain name user
+    * **status**: indicate if an address is available
+    * **reason**: holds error messages if there are any 
+
+##### Hostname and VIP generation
+
+Currently, Kuma allocates a domain for an ExternalService based on the kuma.io/service label and the real destination domain. However, in `MeshExternalService`, we aim to discontinue this practice. Instead, we propose utilizing the HostGenerator for domain generation. To facilitate this change, we intend to introduce a new selector called meshExternalServiceSelector, enabling users to specify particular MeshExternalServices that necessitate custom domains. Only MeshOperator should create a HostGenerator policy and when a user needs a specific domain for a MeshExternalService, the user needs to reach out to the MeshOperator. HostGenerator can be created only on GlobalCP and later is synchronized to all zones. More about HostGenerator in [MADR-046](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/046-meshservice-hostname-vips.md?plain=1#L58).
+
+```yaml
+kind: HostnameGenerator
+metadata:
+  name: k8s-zone-hostnames
+spec:
+  targetRef:
+    kind: MeshExternalService
+    tags:
+      my.label.io/service: my-service 
+  template: {{ name }}.svc.meshext.local
+```
+
+For example, given this `MeshExternalService`
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: mydomain
+  namespace: kuma-system
+  labels:
+    my.label.io/service: my-service 
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+    kuma.io/origin: zone
+spec:
+  type: Managed # Managed|Passthrough|Extension
+  destinations:
+  - address: 192.168.0.1
+    port: 9090 
+...
+```
+
+We would generate such hostname `mydomain.svc.meshext.local`
+
+Possible template functions/keys are:
+
+`{{ name }}` - name of the `MeshExternalService`
+`{{ label "x" }}` - value of label x.
+
+If the template cannot be resolved (label is missing), the hostname won't be generated with a NotAvailable status and an appropriate reason.
+
+#### Positive Consequences
+
+* Clearer structure.
+* Enhanced functionalities.
+* Better policy matching capabilities.
+* No need for a HostnameGenerator
+
+#### Negative Consequences
+
+* Whole new resource with a lot of new code.
+* Temporarily increased complexity of a product until the migration is done.
+* Problem with supporting Universal without transparent proxy, because one MeshExternalService points to many real domains
+* Problem with matching and applying policy when type `Passthrough`.
```

Comment:
can we be more specific here, we can't apply policies on passthrough because the cluster type is original_dst and these configuration options are not available there, right?